stl/inc/yvals_core.h

```diff
@@ -61,6 +61,7 @@
 // P1164R1 Making create_directory() Intuitive
 // P1165R1 Consistently Propagating Stateful Allocators In basic_string's operator+()
 // P1902R1 Missing Feature-Test Macros 2017-2019
+// P1951R1 Default Template Arguments For pair's Forwarding Constructor

```

Comment:
Because (1) this isn't deeply entangled with the existing implementation and (2) it can emit new compiler warnings (I believe that @timsong-cpp's analysis in https://github.com/microsoft/STL/pull/2009#discussion_r657525185 is correct and that these warnings are essentially unavoidable given the compiler's current design), I believe that this feature should be guarded for C++23.

Aside from the C++17 features that we implemented before `/std:c++14` was available, we typically avoid implementing features unconditionally. The ones listed above provide good examples of when to make exceptions: `noexcept` strengthening is already conformant in earlier Standards, the `enable_shared_from_this` overhaul is deeply entangled (and also a very welcome non-breaking cleanup; it may permit some code to compile that technically shouldn't have before, but the old wording was so vague that reasoning about it was extremely difficult), and the `operator+()` allocator propagation was very obscure to observe.